acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -372,9 +372,12 @@ def connect_pcp_client(broker)
     })
     connected = client.connect(5)
     retries += 1
+
+    # If the connection was not established, close it. Otherwise we can end up with 2 successful connections
+    # and get an earlier attempt superseding later connections.
+    client.close if !connected
```

Comment:
On inspecting https://github.com/puppetlabs/ruby-pcp-client/blob/master/lib/pcp/client.rb#L222, https://github.com/faye/faye-websocket-ruby/blob/master/lib/faye/websocket/api.rb#L79, and https://github.com/faye/websocket-driver-ruby/blob/master/lib/websocket/driver.rb#L126, I believe this will not throw an exception.